internal/cli/alpha/module/module.go

```diff
@@ -0,0 +1,118 @@
+package module
+
+import (
+	"context"
+	"fmt"
+	"github.com/kyma-project/cli/internal/kube"
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/utils/pointer"
+	"time"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+var kymaResource = schema.GroupVersionResource{
+	Group:    "operator.kyma-project.io",
+	Version:  "v1alpha1",
+	Resource: "kymas",
+}
+
+type ModulesList []interface{}
+
+type Interactor struct {
+	Client       kube.KymaKube
+	ResourceName types.NamespacedName
+	ForceUpdate  bool
+}
+
+func NewInteractor(client kube.KymaKube, name types.NamespacedName, forceUpdate bool) Interactor {
+	return Interactor{
+		Client:       client,
+		ResourceName: name,
+		ForceUpdate:  forceUpdate,
+	}
+}
+
+func (i *Interactor) Get(ctx context.Context) (ModulesList, error) {
+	namespace := i.ResourceName.Namespace
+	name := i.ResourceName.Name
+	kyma, err := i.Client.Dynamic().Resource(kymaResource).Namespace(namespace).Get(
+		ctx, name, metav1.GetOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("could not get Kyma with name %s and namespace %s: %w", name, namespace, err)
+	}
+
+	modules, _, err := unstructured.NestedSlice(kyma.UnstructuredContent(), "spec", "modules")
+	if err != nil {
+		return nil, fmt.Errorf("could not parse modules spec: %w", err)
+	}
+	return modules, nil
+}
+
+func (i *Interactor) Update(ctx context.Context, modules ModulesList) error {
+	namespace := i.ResourceName.Namespace
+	name := i.ResourceName.Name
+	kyma, err := i.Client.Dynamic().Resource(kymaResource).Namespace(namespace).Get(
+		ctx, name, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("could not get Kyma %s/%s: %w", namespace, name, err)
+	}
+
+	err = unstructured.SetNestedSlice(kyma.Object, modules, "spec", "modules")
+	if err != nil {
+		return fmt.Errorf("failed to set modules list in Kyma spec: %w", err)
+	}
+	kyma.SetManagedFields(nil)
+	data, err := runtime.Encode(unstructured.UnstructuredJSONScheme, kyma)
+	if err != nil {
+		return fmt.Errorf("failed to marshal kyma object: %w", err)
+	}
+	patchOpts := metav1.PatchOptions{FieldManager: "kyma"}
```

Comment:
is this the same field manager as in the k8s.Apply() call? If so, maybe we should think about using a better field manager like "cli.kyma-project.io". WDYT? Other than that looks good!

Reply:
Yes, was wondering why manifests are applied with that generic fieldmanager name. guess `kyma-cli` is also ok, compared to other examples like `kubectl-edit` or `manager`?